examples/basic/main.tf

```diff
@@ -2,46 +2,41 @@ provider "azurerm" {
   features {}
 }
 
-resource "azurerm_resource_group" "example" {
-  name     = "example-resources"
-  location = "West Europe"
+resource "azurerm_resource_group" "rg-connectivity-01" {
+  name     = "rg-connectivity-01"
+  location = "westeurope"
 }
 
-resource "azurerm_network_security_group" "example" {
-  name                = "acceptanceTestSecurityGroup1"
-  location            = azurerm_resource_group.example.location
-  resource_group_name = azurerm_resource_group.example.name
-
-  security_rule {
-    name                       = "test123"
-    priority                   = 100
-    direction                  = "Inbound"
-    access                     = "Allow"
-    protocol                   = "Tcp"
-    source_port_range          = "*"
-    destination_port_range     = "*"
-    source_address_prefix      = "*"
-    destination_address_prefix = "*"
-  }
-
-  tags = {
-    environment = "Production"
-  }
+resource "azurerm_resource_group" "bastion" {
+  name     = "rg-ManagementBastion-prd-01"
+  location = "westeurope"
 }
 
-resource "azurerm_subnet" "example" {
-  name                 = "bn-vnet-01"
-  resource_group_name  = azurerm_resource_group.example.name
-  virtual_network_name = azurerm_virtual_network.vnet.name
-  address_prefixes     = ["0.0.0.0/26"]
+resource "azurerm_virtual_network" "example" {
+  name                = "examplevnet"
+  address_space       = ["192.168.33.0/24"]
+  location            = azurerm_resource_group.rg-connectivity-01.location
+  resource_group_name = azurerm_resource_group.rg-connectivity-01.name
+  depends_on = [
+    azurerm_resource_group.rg-connectivity-01
+  ]
 }
 
 module "azureBastion" {
   source                      = "../../"
-  location                    = "West Europe"
+  location                    = "westeurope"
   state                       = "dev"
   rg_connectivity_name        = "rg-connectivity-01"
-  rg_management_name          = "rg-management-01"
-  subnet_data                 = azurerm_subnet.example
-  network_security_group_data = azurerm_network_security_group.example
+  rg_bastion_name             = azurerm_resource_group.bastion.name
+  bastion_vnet_name           = azurerm_virtual_network.example.name
+  bastion_subnet_address      = ["192.168.33.192/26"]
+  bastion_nsg_name            = "Bastion_NSG"
+  sku                         = "Standard"
+  
+  depends_on = [

```

Comment:
Explicit "depends_on" blocks are not needed when the resource uses an output of a resource. For example the reference above "azurerm_resource_groups.bastion.name" already creates an implicit depency to the RG